language-support/ts/daml-ledger/index.test.ts

```diff
@@ -57,48 +59,57 @@ afterEach(() => {
 describe("streamQuery", () => {
   // we only check that nothing crashes.
   test("receive unknown message", async () => {
+    const restoreConsole = mockConsole();
     const server = new WS("ws://localhost:4000/v1/stream/query", {
       jsonProtocol: true,
     });
     const ledger = new Ledger(mockOptions);
     const stream = ledger.streamQuery(Foo);
     await server.connected;
     console.log("connected");
-    stream.on("change", (evs) => console.log(evs));
-    stream.on("close", (ev) => console.log(ev));
+    stream.on("change", evs => console.log(evs));
+    stream.on("close", ev => console.log(ev));
     server.send("mickey mouse");
+    expect(console.error).toHaveBeenCalledWith("Ledger.streamQuery unknown message", "mickey mouse");
+    restoreConsole();
   });
 
   // we only check that nothing crashes.
   test("receive warnings", async () => {
+    const restoreConsole = mockConsole();
     const server = new WS("ws://localhost:4000/v1/stream/query", {
       jsonProtocol: true,
     });
     const ledger = new Ledger(mockOptions);
     const stream = ledger.streamQuery(Foo);
     await server.connected;
     console.log("connected");
-    stream.on("change", (evs) => console.log(evs));
-    stream.on("close", (ev) => console.log(ev));
-    stream.off("change", (evs) => console.log(evs));
-    stream.off("close", (ev) => console.log(ev));
+    stream.on("change", evs => console.log(evs));
+    stream.on("close", ev => console.log(ev));
+    stream.off("change", evs => console.log(evs));
+    stream.off("close", ev => console.log(ev));
```

Comment:
What are you trying to achieve here? `stream.off` is used to deregister event handlers, which we definitely don't want to do here. Even if we would want to do it, this wouldn't do it since functions are tested for equality by reference and the lambdas in the `off` calls are newly allocated function objects with different references than those in the `on` calls.

If I'm not mistaken, we don't want to call the `off` method of `Stream` anywhere in this test.

Reply:
yes exactly, so I reroute them to the console such that I would see if they were mistakenly called by anything.

Reply:
I make them fail now explicitly.

Reply:
looks like `off` and `on 'close'` is called maybe during teardown of the ledger class, so I can't make it fail.

Reply:
The `on` function is used to register an event handler and the `off` function is used to de-register handlers. I don't think we want to de-register handlers here since that would make the events go nowhere afterwards.

Just FYI, the pattern to register and de-register handlers goes like
```typescript
const onChange = event => { ... };
stream.on("change", onChange);
...
// Later, when we dont' want to learn about "change" events anymore
stream.off("chage", onChange);
```
It is really important to bind the handler to a name, `onChange` here, so that you can use the same reference in the `on` and `off` calls. Otherwise, the `off` call will simply do nothing. Does that make sense?

Reply:
ah ok, my assumption was that the `off` registers a handler for cleanup. I removed those additional lines.